tests/taskcluster/bin/raptor_graph

```diff
@@ -52,36 +63,71 @@ if testing a PULL_REQUEST 'patch build' will be the pr/
 provided HEAD_REV; else it will be BASE_REV -1 commit
 */
 
-var config = {
-  baseTask1: {
-    definition: 'raptor_launch',
-    params: {
-      raptorSuiteName: 'Settings Launch Test (on base)',
-      gaiaRev: process.env.GITHUB_BASE_REV,
-      raptorRuns: 30,
-      raptorApps: 'settings',
-      treeherderSym: 'se1'
-    }
-  },
-  patchTask1: {
-    definition: 'raptor_launch',
-    params: {
-      raptorSuiteName: 'Settings Launch Test (on base + pr)',
-      gaiaRev: process.env.GITHUB_HEAD_REV,
-      raptorRuns: 30,
-      raptorApps: 'settings',
-      treeherderSym: 'se2'
-    }
-  },
-  resultTask: {
+// build the task config object for each app
+// each app will have 1 base task and 1 patch task x TEST_ITERATIONS,
+// and there is always just one result task (per app) that is
+// dependent upon all the base and patch tasks completed
+var config = {};
+
+LAUNCH_TEST_APPS.map(function(nextApp) {
+  // results task requires all base and patch tasks
+  requiredBaseTasks = [];
+  requiredPatchTasks = [];
+
+  // base tasks
+  for (var iteration = 1; iteration <= TEST_ITERATIONS; iteration++) {
+    suiteName = nextApp[0].toUpperCase() + nextApp.slice(1) + " app launch test (on base)";
+    symbol = nextApp.substr(0, 2) + iteration;
+    taskName = nextApp + "BaseTask" + iteration;
+    requiredBaseTasks.push(taskName);
+    config[taskName] = {
+      definition: 'raptor_launch',
+      build: 'base',
+      params: {
+        raptorSuiteName: suiteName,
+        gaiaRev: process.env.GITHUB_BASE_REV,
+        raptorRuns: LAUNCH_TEST_RUNS,
+        raptorApps: nextApp,
+        treeherderSym: symbol
+      }
+    };
+  };
+
+  // patch tasks
+  for (var iteration = 1; iteration <= TEST_ITERATIONS; iteration++) {
```

Comment:
Thanks Greg. The reason for the two iterations is so that all of the base tasks will be made first, followed by the patch tasks. That way when the tasks are actually extended they will be in the order se1, se2, se3... otherwise they would be extended in the order se1, se7, se2, se8, etc. Thanks!